planner/core/planbuilder.go

```diff
@@ -3009,6 +3053,10 @@ func parseAnalyzeOptionsV2(opts []ast.AnalyzeOpt) (map[ast.AnalyzeOptionType]uin
 	if sampleNum > 0 && sampleRate > 0 {
 		return nil, errors.Errorf("You can only either set the value of the sample num or set the value of the sample rate. Don't set both of them")
 	}
+
+	if optMap[ast.AnalyzeOptCMSketchWidth]*optMap[ast.AnalyzeOptCMSketchDepth] > CMSketchSizeLimit {
+		return nil, errors.Errorf("cm sketch size(depth * width) should not larger than %d", CMSketchSizeLimit)
+	}
```

Comment:
Stats v2 doesn't have cm sketch anymore, so this is no longer a necessary check.
I think we don't need this error, or tell the user that this option is useless in the error message.

Reply:
Addressed. See https://github.com/pingcap/tidb/pull/46044/commits/3c399bfe47411e914eae369dfc711da5bfb13029

I removed that check and also checked the version first in handleAnalyzeOptions.